### ISSUE TITLE
Minor cosmetic issue with items created by spells

### ISSUE THREAD

**Issue by Wyreth** [TrinityCore/TrinityCore#20466]
**Description:**

TL;DR: some items get the "Created By" tag when they shouldn't. Some items appear in chat log as "Created item" instead of "Received item" in cases where they shouldn't.

More details:

Currently, almost all the items created by spells are given the "Created By" tag and show the "Created Item X" in chat. This is handled [here](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Spells/SpellEffects.cpp#L1610)

However, only some items created via professions should have that. Items like [Benediction](https://youtu.be/rb2LhTzv-eY?t=440) and [Eternal Fire](https://youtu.be/U4YPNFpuFR8?t=24) shouldn't, yet in TC they both show that behaviour.

Apparently, only profession-creaftable items that aren't consumables and aren't obtainable elsewhere have the "Created By" tag. Gems created by Alchemy don't, sockets created by jewelcrafters do. 

And only player class spells (Conjure Food, Healthstone, Soul Drain) and profession spells should trigger the "Created item <x>" chat message. Everything else should show the "Received item <x>" chat message.

I don't know where leahters crafted with Leatherworking and things like Engineering intermediary items fall in.

Maybe there's some flag for this behaviour, but I haven't been able to find anything yet.

**Branch(es):** both.

**TC rev. hash/commit:** 3994286

**TDB version:** 335.63

**Comment by Treeston** [comment]
Hmm... how about this classification.

Items created with `CREATE_ITEM` that are either:

* Conjured items
* Equippable

should have a `<created by>` tag. All other items should not.

PS: Benediction does not use `CREATE_ITEM`

**Comment by Shauren** [comment]
Hmm thats not it, example
enchanting rods: get <created> tag
engineering gnomish army knife: no tag

**Comment by Aokromes** [comment]
humm maybe related to that flag: tailoring stackable things have made by thing and they don't must have.

**Comment by illfated** [comment]
Same goes for Leatherworking, creating cured hides or higher leather from lower leather also adds the green text `<Made by (Name)>` to the items.
